src/routes/UserSettings/EditProfileTab/EditProfileTab.js

```diff
@@ -13,28 +13,99 @@ import { DEFAULT_BANNER } from 'store/models/Me'
 
 import './EditProfileTab.scss'
 
-const { object, func } = PropTypes
-
-const twitterPrompt = () => window.prompt('Please enter your twitter name.')
-export function linkedinPrompt (prompt) {
-  var url = window.prompt(prompt || 'Please enter the full url for your LinkedIn page.')
-  if (url) {
-    if (validateLinkedin(url)) {
-      return url
-    } else {
-      return linkedinPrompt('Invalid url. Please enter full url for your LinkedIn page.')
+const { object, func, string } = PropTypes
+
+export const socialLinkClicked = provider => {}
+
+const mapSocialProviderToKey = {
+  'twitter': 'twitterName',
+  'linkedin': 'linkedinUrl',
+  'facebook': 'facebookUrl'
+}
+
+/** LinkedIn Url */
+export const validateLinkedinUrl = url => url.match(/^(http(s)?:\/\/)?([\w]+\.)?linkedin\.com/)
+
+export const linkedinPrompt = () => {
+  let linkedinUrl = window.prompt('Please enter the full url for your LinkedIn page.')
+
+  if (linkedinUrl) {
+    while (!validateLinkedinUrl(linkedinUrl)) {
+      linkedinUrl = window.prompt('Invalid url. Please enter the full url for your LinkedIn page.')
     }
+
+    return linkedinUrl
   }
 }
+export  class SocialControl extends Component {
+  static propTypes = {
+    label: string,
+    provider: string,
+    value: string | null,
+    updateSocialSettings: func,
+    handleUnlink: func
+  }
 
-export const validateLinkedin = url => url.match(/^(http(s)?:\/\/)?([\w]+\.)?linkedin\.com/)
+  handleLinkClick () {
+    const { provider, updateSocialSetting } = this.props
 
+    switch (provider) {
+      case 'twitter': {
+        const twitterHandle = window.prompt('Please enter your twitter name.')
+        if (twitterHandle) {
+          updateSocialSetting({ key: mapSocialProviderToKey[provider], value: twitterHandle })
+        }
+        break;
+      }
+      case 'linkedin': {
+        const linkedinUrl = linkedinPrompt()
+        updateSocialSetting({ key: mapSocialProviderToKey[provider], value: linkedinUrl })
+      }
+      // case 'facebook': {
+      //   return onLink()
+      //   .then(({ error }) => {
+      //     if (error) return onChange(false)
+      //     return onChange(true)
+      //   })
+      // }
```

Comment:
TODO: Enable users to save Facebook information

Reply:
not sure how to get FB working locally right now... hmm. one thing people have asked for is the ability to just paste i n a FB URL as an option. do you think you could implement that easily (while leaving in the ability to connect to FB too)? would you need some UI help or could you make a go of it based on whats already there?

Reply:
How do you feel about something like this? 

<img width="427" alt="Screen Shot 2021-04-14 at 10 54 26 AM" src="https://user-images.githubusercontent.com/38114845/114758839-448d1180-9d12-11eb-8c7a-a77be13451cd.png">

`Connect` opens the connect facebook modal

`Link` opens a prompt
<img width="459" alt="Screen Shot 2021-04-14 at 10 54 33 AM" src="https://user-images.githubusercontent.com/38114845/114758953-69818480-9d12-11eb-8f0f-cf26f7a40f81.png">

Reply:
i think thats great for now! fixing FB connect is a separate issue